### PR TITLE
docs: Updates commands in quickstart

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit updates the current Zenith quickstart to use the new `dagger call` syntax. It is superseded by #6262 .

### CODE REVIEW

**Comment by vikram-dagger** [comment]
@sipsma I ran through these commands and came across 4 function calls that fail with the example app. In all cases, the error is the same

```
┃ npm ERR! enoent ENOENT: no such file or directory, open '/src/package.json'                  
┃ npm ERR! enoent This is related to npm not being able to find a file.      
```

Could you please have a look? It may be that the example module code in https://github.com/dagger/hello-dagger needs an update. The failing commands are marked inline.

File: docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+# Quickstart
+
+## Export function outputs
+
+The `export` API allows you to retrieve file, directory and container outputs from functions and write them to the host filesystem.
+
+To see it in action, use `export` with the `build` function used previously, to export the built application to an `out` directory on the host:
+
+```shell
+dagger call build export --path ./out
```
Comment (vikram-dagger) [inline docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx:16]: Fails

File: docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+┃ Asset exported to "./out"
+$ ls out/
+asset-manifest.json  index.html   logo512.png    robots.txt
+favicon.ico          logo192.png  manifest.json  static
+```
+
+The `export` API also works with functions that return `Container` types, in which case the `Container` is exported as a TAR file to the host file system. Here's an example:
+
+```shell
+dagger call app-container export --path ./container.tar
```
Comment (vikram-dagger) [inline docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx:32]: Fails

File: docs/versioned_docs/version-zenith/user/quickstart/302010-shell.mdx [hunk 3]
```diff
@@ -32,12 +30,12 @@ Linux vtmeopf0atpc2.3jgto8svt9r6s.dagger.local 6.2.0-34-generic #34-Ubuntu SMP P
...
 
-By default, `dagger shell` will execute the container's entrypoint. If the container doesn't have an entrypoint, it will default to executing `sh`.
+By default, `shell` will execute the container's entrypoint. If the container doesn't have an entrypoint, it will default to executing `sh`.
 
-However, you can also override the command that is executed when the shell session starts via the `--entrypoint` flag. You can use this to (for example) drop into a `node` shell in the container returned by `debug`, rather than `sh`. Use the following command to understand how this works:
+However, you can also override the command that is executed when the shell session starts via the `--args` argument. You can use this to (for example) drop into a `node` shell in the container returned by `debug`, rather than `sh`. Use the following command to understand how this works:
 
 ```shell
-dagger shell --entrypoint node debug
+dagger call debug shell --args node
```
Comment (vikram-dagger) [inline docs/versioned_docs/version-zenith/user/quickstart/302010-shell.mdx:38]: Fails

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx [hunk 4]
```diff
@@ -21,7 +21,7 @@ In order for this to work, the service returned by the function must have the `C
 Use the `dagger up` command to build the React application and serve it with NGINX:
 
 ```shell
-dagger up --native service
+dagger call service up --native
```
Comment (vikram-dagger) [inline docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:24]: Fails

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx [hunk 5]
```diff
@@ -33,14 +33,14 @@ The `--native` flag results in the exposed ports on the container being auto-map
 However, you can modify this behavior and choose a different port on your host via the `--port` flag. For example:
 
 ```shell
-dagger up --port 9090:8080 service
+dagger call service up --ports 9090:8080
```
Comment (vikram-dagger) [inline docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:36]: Fails

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx [hunk 6]
```diff
@@ -33,14 +33,14 @@ The `--native` flag results in the exposed ports on the container being auto-map
...
 ```
 
 This results in container port 8080 being mapped to host port 9090. You should now be able to access the application by browsing to `http://localhost:9090` (or equivalent host name) on the host.
 
 :::tip
 Here is an example of using this command with a module from the [Daggerverse](https://daggerverse.dev/):
 
-- `dagger up --native --mod github.com/helderco/daggerverse/python@kcna23 http-server --src <path/to/web/assets>`
+- `dagger call -m github.com/helderco/daggerverse/python@kcna23 http-server up --src <path/to/web/assets> --native`
```
Comment (vikram-dagger) [inline docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:44]: Fails, for @helderco 
```
┃ ImportError: cannot import name 'Annotated' from 'dagger.mod' 
```

**Comment by gerhard** [comment]
I **think** that we want to merge #6262 first, then this one. OK to do the same here too: https://github.com/dagger/dagger/pull/6262#issuecomment-1921463759

**Comment by vikram-dagger** [comment]
> I **think** that we want to merge #6262 first, then this one. OK to do the same here too: [#6262 (comment)](https://github.com/dagger/dagger/pull/6262#issuecomment-1921463759)

If #6262 is merged, then this PR becomes irrelevant. In case it's not, then we should merge this one because it does a surface update to the current quickstart for the new `dagger call` syntax, to ensure that docs are not broken on release.

**Comment by helderco** [comment]
@vikram-dagger, I think the failures you're getting mean the hello-dagger repo needs fixing. Possibly this dependency: https://github.com/dagger/hello-dagger/blob/438911579b2e8ee2e1d2172ee347630120553d4a/dagger.json#L5

```
json: error calling MarshalJSON for type *main.Directory: input: resolve: node: withVersion: withNpm: withSource: install: build: container: directory: process "npm install" did not complete successfully: exit code: 254
input: resolve: node: withVersion: withNpm: withSource: install: build: container: directory: process "npm install" did not complete successfully: exit code: 254
```

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx (hunk 6 above)
Comment (helderco) [reply docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:44]: Remove `@kcna23`. That's fixed to last kubecon.

Comment (helderco) [reply docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:44]: Also, `up` is in the wrong place. `--src` is for `http-server` function:
```suggestion
- `dagger call -m github.com/helderco/daggerverse/python http-server --src <path/to/web/assets>  up --native`
```

File: docs/versioned_docs/version-zenith/user/quickstart/302010-shell.mdx (hunk 3 above)
Comment (vikram-dagger) [reply docs/versioned_docs/version-zenith/user/quickstart/302010-shell.mdx:38]: Fixed now

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx (hunk 6 above)
Comment (vikram-dagger) [reply docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:44]: Fixed

**Comment by vikram-dagger** [comment]
> @vikram-dagger, I think the failures you're getting mean the hello-dagger repo needs fixing. Possibly this dependency: https://github.com/dagger/hello-dagger/blob/438911579b2e8ee2e1d2172ee347630120553d4a/dagger.json#L5

It wasn't the dependency but I found and fixed the issue: https://github.com/dagger/hello-dagger/commit/59baf44aa28e78481ca4ecc3a01b4d69ff255dd3
 
> There's also the changes to `dagger.json`.

I've added these as well, thanks!

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx (hunk 5 above)
Comment (vikram-dagger) [reply docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:36]: Fixed now

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx (hunk 4 above)
Comment (vikram-dagger) [reply docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:24]: Fixed now

File: docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx (hunk 1 above)
Comment (vikram-dagger) [reply docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx:16]: Fixed now

File: docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx (hunk 2 above)
Comment (vikram-dagger) [reply docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx:32]: Yes, I did so already https://github.com/dagger/dagger/pull/6548#issuecomment-1921052528 not

Comment (vikram-dagger) [reply docs/versioned_docs/version-zenith/user/quickstart/292861-export.mdx:32]: Fixed now

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx (hunk 6 above)
Comment (sipsma) [reply docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx:44]: @vikram-dagger did you forget to merge the fixes here? The diff I'm seeing for the PR still shows `dagger call -m github.com/helderco/daggerverse/python@kcna23 http-server up --src <path/to/web/assets> --native`

**Review by sipsma — APPROVED** [review]

File: docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx (hunk 6 above)
Comment (helderco) [reply docs/versioned_docs/version-zenith/user/quickstart/532928-up.mdx]: I see it.

**Review by helderco — APPROVED** [review]

**Pull request merged**